src/states.js

```diff
@@ -52,41 +49,38 @@ class State {
     throw new TLSError(ALERT_DESCRIPTION.UNEXPECTED_MESSAGE);
   }
 
-  async recvAlertMessage(err) {
-    switch (err.description) {
-      case ALERT_DESCRIPTION.USER_CANCELLED:
-        // It's not obvious from the spec whether we should do anything here,
-        // or just wait for the following `CLOSE_NOTIFY` alert.
-        break;
+  async recvAlertMessage(alert) {
+    switch (alert.description) {
       case ALERT_DESCRIPTION.CLOSE_NOTIFY:
-        this.conn._closeForRecv();
-        break;
+        this.conn._closeForRecv(alert);
+        throw alert;
```

Comment:
Basically, yes.  I think a better approach is probably to shuffle the API to use an `EventEmitter` style, so it can emit a `close` event in this case, which is basically what the higher-level `PairingChannel` abstraction does anyway.  But it was a bigger refactor than I wanted to take on inline in this PR.